back/boxtribute_server/cli/service.py

```diff
@@ -0,0 +1,161 @@
+from auth0 import Auth0Error
+from auth0.authentication import GetToken
+from auth0.management import Auth0
+
+from ..exceptions import ServiceError
+from .utils import setup_logger
+
+LOGGER = setup_logger(__name__)
+
+
+class Auth0Service:
+    def __init__(self, interface):
+        self._interface = interface
+
+    def get_users_of_base(self, base_id):
+        """Fetch all users of the given base. Return lists of users sorted by
+        single-base/multi-base type.
+        """
+        base_id = str(base_id)
+        # https://github.com/auth0/auth0-python/blob/6b1199fc74a8d2fc6655ffeef09ae961dc0b8c37/auth0/management/users.py#L55
+        users = []
+        try:
+            # Pagination setup
+            page = 0
+            per_page = 50
+            rest = None
+            while True:
+                response = self._interface.users.list(
+                    q=f'app_metadata.base_ids:"{base_id}"',
+                    fields=["app_metadata", "user_id", "name", "blocked"],
+                    page=page,
+                    per_page=per_page,
+                )
+                LOGGER.info(
+                    f"Fetched page {page + 1} of user data of total {response['total']}"
+                    " users."
+                )
+                if rest is None:
+                    rest = response["total"]
+                users.extend(response["users"])
+
+                # Pagination logic: go to next page; stop if nothing left
+                page += 1
+                rest -= per_page
+                if rest < 1:
+                    break
+        except Auth0Error as e:
+            raise ServiceError(code=e.status_code, message=e.message)
+
+        # Sort response into single- and multi-base users
```

Comment:
TODO: please add a comment that we need to revisit this section when we remove base information from app_metadata